packages/patternfly-4/react-core/src/components/TextArea/TextArea.tsx

```diff
@@ -0,0 +1,58 @@
+import * as React from 'react';
+import { HTMLProps } from 'react';
+import styles from '@patternfly/patternfly/components/FormControl/form-control.css';
+import { css } from '@patternfly/react-styles';
+import { Omit } from '../../helpers/typeUtils';
+
+export interface TextAreaProps extends Omit<HTMLProps<HTMLTextAreaElement>, 'onChange'> {
+  /** Additional classes added to the TextArea. */
+  className?: string;
+  /** Flag to show if the TextArea is required. */
+  isRequired?: boolean;
+  /** Flag to show if the TextArea is valid or invalid. */
+  isValid?: boolean;
+  /** Value of the TextArea. */
+  value?: string | number;
+  /** A callback for when the TextArea value changes. */
+  onChange?: (value: string, event: React.ChangeEvent<HTMLTextAreaElement>) => void;
+  /** Custom flag to show that the TextArea requires an associated id or aria-label. */
+  'aria-label'?: string;
+}
+
+
+export class TextArea extends React.Component<TextAreaProps> {
+
+  static defaultProps = {
+    className: '',
+    isRequired: false,
+    isValid: true,
+    'aria-label': null as string,
+  }
+  
+
+  constructor(props: TextAreaProps) {
+    super(props);
+    if (!props.id && !props['aria-label']) {
+      // tslint:disable-next-line:no-console
+      console.error('TextArea: TextArea requires either an id or aria-label to be specified');
+    }
+  }
+
+  private handleChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+      this.props.onChange(event.currentTarget.value, event);
```

Comment:
`onChange` needs a default; this will throw a console error if it's called while `onChange` is `undefined`.